### DISCUSSION TITLE
Get Link details by URL

### DISCUSSION THREAD

**Discussion by prplecake** [Kovah/LinkAce#656] [Ideas & Feature Requests]
Hi!

I'm working on a browser extension for LinkAce, which I'm modeling pretty heavily after Pinboard+ right now. One of the features from Pinboard+ is the ability to update or delete links from the extension. This is possible because the Pinboard API allows you to get information on a link if it exists.

Right now to implement similar functionality, I'd have to get all links and grep that for the current URL, which seems wasteful.

Let me know if you're open to this, if I can find time in the coming weeks I might even be able to submit a PR.

**Comment by Kovah** [comment]
If you're using the API, you could use the search endpoint with a full URL. If the result is empty, the link does not exists.


```bash
curl --request GET \
  --url 'https://demo.linkace.org/api/v1/search/links?query=https%3A%2F%2Fdev.to%2Fkovah%2Fhow-i-built-a-web-tool-for-gamers-3hp' \
  --header 'Accept: application/json' \
  --header 'Authorization: Bearer s9anum7ODLsxbMQCr4WnYSahCEZViFEB'
```

See https://api-docs.linkace.org/#tag/search

**Reply by prplecake** [reply]
Ah, admittedly I didn't try that endpoint since the description of the query parameter didn't seem to do what I needed. The description is "Query used to search for Tags", I suppose that should be "Query used to search for Links".

Anyway, great, that should work for me. Thanks!

**Reply by Kovah** [reply]
Great catch, will fix that right away.

Btw cool avatar. Todd was so great.
